### ISSUE TITLE
XDG Base Directory Specification not followed

### ISSUE THREAD

**Issue by maresb** [FreeTDS/freetds#400]
According to the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), I would expect to be able to place my `~/.freetds.conf` file instead in `$HOME/.config/freetds/freetds.conf`, but this is currently unsupported by FreeTDS. (More generally, it should look in `$XDG_CONFIG_HOME/freetds/freetds.conf` when `XDG_CONFIG_HOME` is defined.)

Would you be open to supporting this? Specifically, searching for the config file inside `$XDG_CONFIG_HOME` or `$HOME/.config` before looking for `~/.freetds.conf`? Thanks!

**Comment by fziglio** [comment]
Not that freetds it's a desktop application. Yes, I would like to have `~/.config/freetds` instead of the old configurations all in the home, it would make sense. About the `XDG_CONFIG_HOME` freetds don't even trust `HOME` environment and use `getpwuid` (or system equivalent, `HOME` is used only as a last resort).

Well, not that it cannot be overwritten by many environment variables at the moment (like `FREETDSCONF` or `FREETDS`).

I think all you need is in `src/tds/config.c`.

**Comment by freddy77** [comment]
@maresb are you going to propose a patch for this?

**Comment by maresb** [comment]
No, sorry, I'm not using this anymore. Feel free to close if nobody else is willing to pick this up. Thanks!
